openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -798,6 +799,50 @@ Here is another example JWT not only proving possession of a private key but als
 }
 ```
 
+#### `ldp_vp` Key Proof Type {#ldp_vp-proof-type}
+
+The verifiable presentation as defined by [@VC_DATA_2.0] or [@VC_DATA] signed using Data Integrity that are used as Key Proofs MUST contain the following elements:
+
+  * in the presentation itself,
+      * `holder`: OPTIONAL. MUST be equivalent to the verificationMethod identifier (e.g. DID) in the `proof.verificationMethod` property.
+
+  * in the proof body, 
+      * `domain`: REQUIRED (string). The value of this claim MUST be the Credential Issuer Identifier.
+      * `challenge`: REQUIRED (string). The value type of this claim MUST be a string, where the value is a server-provided `c_nonce`. It MUST be present when the Wallet received server-provided `c_nonce`.
+
+The Credential Issuer MUST validate that the `proof` is actually signed with a key in possession of the Holder.
```

Comment:
I don't know what to do about this language since it's "the right language / commonly used language" when considering other specifications around "proof of possession". However, this is technically incorrect for more or less all specs that use that terminology (so, not specific to this PR).

In short, "proof of possession" is really "proof of use", as the person who provides this proof does not actually need to "possess" the key / its secret material, they only need to be able to generate a signature over what is required (e.g., the VP above and the challenge, provided via `c_nonce` in OID4VCI). Someone who exposes an authz-protected API to use a particular key could grant anyone they wanted access to that API without ever exposing the private key material -- and a generated proof would therefore not be a "proof of possession" for such a user of this API.

I don't know that we want to take on trying to fix this language here in this one spot since every other "proof of possession" spec has a similar problem. But the Credential Issuer is *not capable* of performing the validation based on the language here -- and the same applies to other language used in other specs around proof of possession. If we want this to be the first place that we try to improve this to better communicate actual guarantees and requirements, here's a suggestion:

```suggestion
The Credential Issuer MUST verify that the `proof` was generated by a key associated with its verification method, thereby demonstrating that the Holder was capable of causing this key to be used to generate the proof.
```

Reply:
Even if it is different in the rest of the spec, i prefer your more precise description here. As you said: The proof of possession is actually a proof of being able to use/access the keys.

But this is rather up to @Sakurann or @tlodderstedt to decide.

Reply:
yes, i would prefer not to make this change in this PR. feel free to raise an issue and address it more systematically if agreed to.